analyze_test.go

```diff
@@ -225,14 +225,40 @@ func TestAnalyze_njs(t *testing.T) {
 			blockCtx{"http", "server", "if"},
 			true,
 		},
+		"js_periodic ok in http": {
+			&Directive{
+				Directive: "js_periodic",
+				Args:      []string{"function"},
+				Line:      5,
+			},
+			blockCtx{"http", "location"},
+			false,
+		},
+		"js_periodic not ok in http": {
```

Comment:
nit, subtest name should better match the test expectation?

```suggestion
		"js_periodic not ok if": {
```